### DISCUSSION TITLE
error creating a transaction: sqlitex.Exec: sqlite: clear bindings: interrupted

### DISCUSSION THREAD

**Discussion by thedmdim** [zombiezen/go-sqlite#70] [Q&A] [answered]
Hi! I use `zombiezen.com/go/sqlite` in my backend and recently in log I noticed a bunch of similar errors `error creating a transaction: sqlitex.Exec: sqlite: clear bindings: interrupted`
occuring in this code:

```go
func (s *Router) StoreVideos(conn *sqlite.Conn, videos map[string]*Video) error {

	endFn, err := sqlitex.ImmediateTransaction(conn)
	if err != nil {
		return fmt.Errorf("error creating a transaction: %w", err)
	}
	defer endFn(&err)

	stmt := conn.Prep("INSERT INTO videos (id, uploaded, title, views, vertical, category) VALUES (?, ?, ?, ?, ?, ?);")
	for _, video := range videos {

		stmt.BindText(1, video.ID)
		stmt.BindInt64(2, video.UploadedAt)
		stmt.BindText(3, video.Title)
		stmt.BindInt64(4, int64(video.Views))
		stmt.BindBool(5, video.Vertical)
		stmt.BindInt64(6, int64(video.Category))

		if _, err := stmt.Step(); err != nil {
			return fmt.Errorf("stmt.Step: %w", err)
		}
		if err := stmt.Reset(); err != nil {
			return fmt.Errorf("stmt.Reset: %w", err)
		}
		if err := stmt.ClearBindings(); err != nil {
			return fmt.Errorf("stmt.ClearBindings: %w", err)
		}
	}
	return nil
}
```

I googled that I had to clear bindings everytime, but as you see there already is `stmt.ClearBindings()` call. The function executes in usual `ServeHTTP` like function. 

Any ideas of how could i fix this?

**Comment by zombiezen** [comment, marked as answer]
The error you're seeing is what happens when a statement is interrupted by its [interrupt channel](https://pkg.go.dev/zombiezen.com/go/sqlite#Conn.SetInterrupt). That's usually set by the pool when you use `Get` with the request's context. These sorts of errors are to be expected.

**Reply by thedmdim** [reply]
Thanks! This is really helpful, i'll use ctx with timeout instead of request's one

**Reply by zombiezen** [reply]
Another option: my [`zombiezen.com/go/xcontext.KeepAlive` function](https://pkg.go.dev/zombiezen.com/go/xcontext#KeepAlive) allows you to use the request context but keep it alive for a minimum time. That way, a permitted long-running request won't interrupt unnecessarily.
